switchtec_dma.c

```diff
@@ -611,14 +613,17 @@ static void switchtec_dma_process_desc(struct switchtec_dma_chan *swdma_chan)
 		spin_lock_bh(&swdma_chan->complete_lock);
 
 		ce = switchtec_dma_get_ce(swdma_chan, swdma_chan->cq_tail);
-		if (le16_to_cpu(ce->phase_tag) == swdma_chan->phase_tag) {
+		phase_tag = smp_load_acquire(&ce->phase_tag);
```

Comment:
I'm not following why phase_tag is now `smp_load_acquire()`'d. I don't see a companion "store_release"; in fact, this field seems to be written by the DMA hardware itself.

Reply:
Yes, phase_tag is updated by the DMA hardware. phase_tag acts as the indicator of CE's readiness for host to consume. I make it `smp_load_acquire()`'d here just to ensure that all the other access to CE must be after the check of phase_tag. And I think in this case a companion store_release is not necessary.

Reply:
Okay. I guess my thinking was that the spin_lock_bh immediately above has implicit acquire semantics, and was wondering if some hardware fence was intended with this change. At any rate, I don't think it's wrong, per se (and outside of x86 I have no instinct for which shade of barrier is required).